### DISCUSSION TITLE
Dinamically load variants or the content of a page depending the user/role

### DISCUSSION THREAD

**Discussion by utamayo** [GitbookIO/community#812] [Feature Requests]
Any possibility to show some sections where are hidden or variant depending on the user who is accessing? I am using Auth0 to authenticate users and It will be great to show some restricted sections to some users only.

**Comment by SamyPesse** [comment]
Not yet, but soon ;)

What data would you like to use conditionally show the variants/sections? Do you store all of these data as Auth0 properties?

**Reply by utamayo** [reply]
I am a Teacher and I would like to load depending the emails of the teachers and students, a exercise page/section with solutions or without solutions
My custom action in Auth0 is this to allow my students and other teacher to enter into the gitbook site:

exports.onExecutePostLogin = async (event, api) => {

  // List of allowed email addresses
  const allowed = [
    'yyyyy.zzz@xxx.com',
    'aaaa.bbbb@xxx.com'
  ];

  // Check if the user's email is in the allowed list
  if (!allowed.includes(event.user.email)) {
    // Deny access if the email is not allowed
    api.access.deny('Access denied.');
  }
};

I don't know if this will help you to clarify my feature request or it could be done simplier but I didn't know how to do it better and simplier.
